### PR TITLE
Feat/navigation view orientation

### PR DESCRIPTION
**Pull request by Archdoog**
- Adds portrait navigation view and outer orienting navigation view.
- Separates navigation view & navigation map view.
- Adds content inset system for navigation view.

### CODE REVIEW

File: apple/Sources/FerrostarMapLibreUI/Views/NavigationMapView.swift [hunk 1]
```diff
@@ -6,15 +6,15 @@ import MapLibreSwiftDSL
 import MapLibreSwiftUI
 import SwiftUI
 
-// TODO: Rename this NavigationView? Basically to detach the 
+// TODO: Rename this NavigationView? Basically to detach the
```
Comment (ianthetechie) [inline apple/Sources/FerrostarMapLibreUI/Views/NavigationMapView.swift:9]: 😂

File: apple/Sources/FerrostarMapLibreUI/Views/NavigationMapView.swift [hunk 2]
```diff
@@ -24,13 +27,13 @@ public struct NavigationMapView: View {
         darkStyleURL: URL,
         navigationState: NavigationState?,
         camera: Binding<MapViewCamera>,
-        distanceFormatter: Formatter = MKDistanceFormatter()
+        @MapViewContentBuilder _ makeMapContent: () -> [StyleLayerDefinition] = { [] }
```
Comment (ianthetechie) [inline apple/Sources/FerrostarMapLibreUI/Views/NavigationMapView.swift:30]: What happened to the distance formatter? We definitely want to keep that user- configurable.

File: apple/Sources/FerrostarMapLibreUI/Views/NavigationMapView.swift [hunk 3]
```diff
@@ -60,55 +63,25 @@ public struct NavigationMapView: View {
...
-        lightStyleURL: URL(string: "https://demotiles.maplibre.org/style.json")!,
-        darkStyleURL: URL(string: "https://demotiles.maplibre.org/style.json")!,
-        navigationState: state,
-        camera: .constant(.center(state.snappedLocation.clLocation.coordinate, zoom: 12)),
-        distanceFormatter: formatter
-    )
-}
-
-#Preview("Navigation Map View (Metric)") {
+#Preview("Navigation Map View") {
```
Comment (ianthetechie) [inline apple/Sources/FerrostarMapLibreUI/Views/NavigationMapView.swift:77]: Similarly, I'd like to keep this as a demo of using multiple locales.

**Review by ianthetechie — COMMENTED** [review]
Sorry GitHub only showed me a partial diff or something at first.

File: apple/Sources/FerrostarMapLibreUI/Views/NavigationMapView.swift (hunk 2 above)
Comment (Archdoog) [reply apple/Sources/FerrostarMapLibreUI/Views/NavigationMapView.swift:29]: This is only a "MapView" now, so there's nothing to format here. It's all pushed up one later to the "Navigation Views" where the instruction view is included.

**Review by ianthetechie — APPROVED** [review]

**Pull request merged**